File: Problems/Algorithms/268. Missing Number/MissingNumber.java
class Solution {
    public int missingNumber(int[] nums) {
        int ans = 0;
        
        for (int num : nums) {
            ans ^= num;
        }
        
        for (int num = 0; num <= nums.length; num++) {
            ans ^= num;
        }
        
        return ans;
    }
}

File: Problems/Algorithms/268. Missing Number/MissingNumber.kt
class Solution {
    fun missingNumber(nums: IntArray): Int {
        var ans = 0
        
        for (num in nums) {
            ans = (ans xor num)
        }
        
        for (num in 0..nums.size) {
            ans = (ans xor num)
        }
        
        return ans
    }
}

